### PR TITLE
Minor fixes from #264 suggested by Martin

### PR DESCRIPTION
**Pull request by richajaindce**

### CODE REVIEW

File: src/protocol/sort/bit_permutation.rs [hunk 1]
```diff
@@ -109,11 +108,7 @@ mod tests {
         let h1_future = bit_permutation(ctx1, shares[1].as_slice());
         let h2_future = bit_permutation(ctx2, shares[2].as_slice());
 
-        let result: [_; 3] = try_join_all([h0_future, h1_future, h2_future])
-            .await
-            .unwrap()
-            .try_into()
-            .unwrap();
+        let result: [_; 3] = join3(h0_future, h1_future, h2_future).await;
```
Comment (martinthomson) [inline src/protocol/sort/bit_permutation.rs:111]:
```suggestion
        let result = join3(h0_future, h1_future, h2_future).await;
```
No need for the type annotation here.

File: src/protocol/sort/bit_permutation.rs [hunk 2]
```diff
@@ -151,11 +146,7 @@ mod tests {
         let h1_future = bit_permutation(mc1.ctx, shares[1].as_slice());
         let h2_future = bit_permutation(mc2.ctx, shares[2].as_slice());
 
-        let result: [_; 3] = try_join_all([h0_future, h1_future, h2_future])
-            .await
-            .unwrap()
-            .try_into()
-            .unwrap();
+        let result: [_; 3] = join3(h0_future, h1_future, h2_future).await;
```
Comment (martinthomson) [inline src/protocol/sort/bit_permutation.rs:149]:
```suggestion
        let result = join3(h0_future, h1_future, h2_future).await;
```

File: src/protocol/sort/compose.rs [hunk 3]
```diff
@@ -85,15 +85,12 @@ mod tests {
             let world: TestWorld = make_world(QueryId);
             let [ctx0, ctx1, ctx2] = make_contexts(&world);
 
-            let sigma_and_randoms: [_; 3] = try_join_all([
+            let sigma_and_randoms: [_; 3] = join3(
```
Comment (martinthomson) [inline src/protocol/sort/compose.rs:88]:
```suggestion
            let sigma_and_randoms = join3(
```

File: src/protocol/sort/compose.rs [hunk 4]
```diff
@@ -123,11 +120,7 @@ mod tests {
                 rho2,
             );
 
-            let result: [_; 3] = try_join_all([h0_future, h1_future, h2_future])
-                .await
-                .unwrap()
-                .try_into()
-                .unwrap();
+            let result: [_; 3] = join3(h0_future, h1_future, h2_future).await;
```
Comment (martinthomson) [inline src/protocol/sort/compose.rs:123]:
```suggestion
            let result = join3(h0_future, h1_future, h2_future).await;
```

File: src/protocol/sort/secureapplyinv.rs [hunk 5]
```diff
@@ -90,15 +90,12 @@ mod tests {
 
             let [perm0, perm1, perm2] = generate_shares::<Fp31>(&permutation);
 
-            let perm_and_randoms: [_; 3] = try_join_all([
+            let perm_and_randoms: [_; 3] = join3(
```
Comment (martinthomson) [inline src/protocol/sort/secureapplyinv.rs:93]:
```suggestion
            let perm_and_randoms = join3(
```

File: src/protocol/sort/secureapplyinv.rs [hunk 6]
```diff
@@ -128,11 +125,7 @@ mod tests {
                 &perm_and_randoms[2].0,
             );
 
-            let result: [_; 3] = try_join_all([h0_future, h1_future, h2_future])
-                .await
-                .unwrap()
-                .try_into()
-                .unwrap();
+            let result: [_; 3] = join3(h0_future, h1_future, h2_future).await;
```
Comment (martinthomson) [inline src/protocol/sort/secureapplyinv.rs:128]:
```suggestion
            let result = join3(h0_future, h1_future, h2_future).await;
```

File: src/protocol/sort/shuffle.rs [hunk 7]
```diff
@@ -284,11 +283,7 @@ mod tests {
         let h1_future = shuffle_shares(shares1, (perm2.0.as_slice(), perm2.1.as_slice()), c1);
         let h2_future = shuffle_shares(shares2, (perm3.0.as_slice(), perm3.1.as_slice()), c2);
 
-        let results: [_; 3] = try_join_all([h0_future, h1_future, h2_future])
-            .await
-            .unwrap()
-            .try_into()
-            .unwrap();
+        let results: [_; 3] = join3(h0_future, h1_future, h2_future).await;
```
Comment (martinthomson) [inline src/protocol/sort/shuffle.rs:286]:
```suggestion
        let results = join3(h0_future, h1_future, h2_future).await;
```

**Review by martinthomson — APPROVED** [review]
Some suggestions that I will apply.

File: src/protocol/sort/generate_permutation.rs [hunk 8]
```diff
@@ -210,44 +213,38 @@ mod tests {
...
+            BATCHSIZE.try_into().unwrap(),
+            perm1,
+        );
+        let h2_future = shuffle_and_reveal_permutation(
+            ctx2.narrow("shuffle_reveal"),
+            BATCHSIZE.try_into().unwrap(),
+            perm2,
+        );
+
+        let perms_and_randoms: [_; 3] = join3(h0_future, h1_future, h2_future).await;
```
Comment (martinthomson) [inline src/protocol/sort/generate_permutation.rs:241]:
```suggestion
        let perms_and_randoms = join3(h0_future, h1_future, h2_future).await;
```

**Comment by akoshelev** [comment]
lmk folks if you want me to merge it

**Comment by benjaminsavage** [comment]
Richa - I think many of these changes have now been merged. Can you rebase?

File: src/protocol/reveal/mod.rs [hunk 9]
```diff
@@ -166,13 +164,20 @@ mod tests {
...
 
+            let results =
+                join3v(zip(v.iter(), m_shares).map(|(v, m_share)| async move {
+                    v.context().reveal(record_id, &m_share).await
+                }))
+                .await;
+
             assert_eq!(input, results[0]);
             assert_eq!(input, results[1]);
             assert_eq!(input, results[2]);
```
Comment (benjaminsavage) [inline src/protocol/reveal/mod.rs:183]: There's a much better way to write tests now. Try out the new hotness:

```
let things_to_reveal = [ /* array of ten randomly generated Fp31s you want to reveal */ ];
let world = TestWorld::new(QueryId);
let result = world
  .malicious(
      things_to_reveal,
      |ctx, m_shares| async move { 
          try_join_all(m_shares.enumerate().map(|(i, m_share) async move {
              ctx.reveal(m_share).await
          })),
      }
  )
  .await;
```

Or something like that.

File: src/protocol/sort/bit_permutation.rs [hunk 10]
```diff
@@ -90,7 +85,12 @@ mod tests {
     pub async fn semi_honest() {
         let world = TestWorld::new(QueryId);
 
-        let result = join3(h0_future, h1_future, h2_future).await;
+        let result = world
+            .semi_honest(
+                INPUT.iter().map(|x| Fp31::from(*x)),
+                |ctx, m_shares| async move { bit_permutation(ctx, &m_shares).await.unwrap() },
+            )
+            .await;
```
Comment (benjaminsavage) [inline src/protocol/sort/bit_permutation.rs:93]: Funny... I already see this code on main. Are you sure you rebased?

**Review by benjaminsavage — APPROVED** [review]

**Pull request merged**